timemachine/md/minimizer.py

```diff
@@ -140,85 +142,181 @@ def minimize_host_4d(
+    non_ligand_group_idxs = [group for group in group_idxs if np.all(group < num_host_atoms)]
+
     u_impl = summed_potential_bound_impl_from_potentials_and_params(potentials, params)
     bound_impls = [u_impl]
-    x = fire_minimize(x0, bound_impls, box, n_steps_per_window)
-
-    # No need to reconstruct the context, just change the bound potential params. Allows
-    # for preserving the velocities between windows
-    ctxt = custom_ops.Context(x, v0, box, intg, bound_impls)
-    for lamb in np.linspace(1.0, 0, windows):
-        _, params = parameterize_system(hgt, ff, lamb)
-        u_impl.set_params(flatten_params(params))
-        xs, _ = ctxt.multiple_steps(n_steps_per_window)
-        x = xs[-1]
-
-    final_coords = fire_minimize(x, bound_impls, box, n_steps_per_window)
+
+    baro = MonteCarloBarostat(
+        x0.shape[0],
+        pressure,
+        temperature,
+        non_ligand_group_idxs,
+        barostat_interval,
+        seed + 1,
+    )
+    baro_impl = baro.impl(bound_impls)
+
+    ctxt = custom_ops.Context(x0, v0, box, intg, bound_impls, movers=[baro_impl])
+    xs, boxes = ctxt.multiple_steps(equilibration_steps)
+    x = xs[-1]
+    box = boxes[-1]
+
+    assert np.all(x[num_host_atoms:] == np.concatenate(mol_coords)), "Ligand atoms unexpectedly moved"
+
+    # Only evaluate the host forces, the mols may be strained at this stage. No change is made to the mols
+    # which means evaluating the mols forces may trigger spurious failures
     for impl in bound_impls:
-        du_dx, _ = impl.execute(final_coords, box, compute_u=False)
-        check_force_norm(-du_dx)
+        du_dx, _ = impl.execute(x, box, compute_u=False)
+        check_force_norm(-du_dx[:num_host_atoms])
+
+    return x[:num_host_atoms], box
+
+
+def fire_minimize_host(
+    mols: List[Chem.Mol],
+    host_config: HostConfig,
+    ff: Forcefield,
+    mol_coords: Optional[List[NDArray]] = None,
+    n_steps_per_window: int = 500,
+    max_lambda: float = 0.1,
+    n_windows: int = 2,
+) -> NDArray:
+    """
+    Minimize a host system using the Fire minimizer.
+
+    The ligand coordinates are fixed during minimization, and only host_coords are minimized.
+
+    Parameters
+    ----------
+    mols: list of Chem.Mol
+        Ligands to be inserted. This must be of length 1 or 2 for now.
+
+    host_config: HostConfig
+        Represents the host system.
+
+    ff: ff.Forcefield
+        Wrapper class around a list of handlers
+
+    mol_coords: list of np.ndarray
+        Pre-specify a list of mol coords. Else use the mol.GetConformer(0)
+
+    n_steps_per_window: integer
+        Number of steps to run the FIRE minimizer with at each window
+
+    max_lambda: float
+        The largest lambda value to run a window at. If the value is 1.0 the mols will be
+        completely decoupled.
+
+    n_windows: integer
+        The number of windows to linearly interpolate between the max_lambda value and 0.0.
```

Comment:
clarification: will this correspond to linear interpolation of `w_coord` between `0` and `max_lambda_value * cutoff`, or some pre-optimized interpolation like https://github.com/proteneer/timemachine/blob/ab3def743ad562c2d89e90b8ea54e989a6de5a2a/timemachine/fe/lambda_schedule.py#L22 ?

(linear probably suffices, but it's just good to note which one is chosen here)

Reply:
Will clarify, it is linear

Reply:
Clarified in https://github.com/proteneer/timemachine/pull/1339/commits/ffdf00f744e046df632ea2b083ba6ce485231d2d